### PR TITLE
Add float support

### PR DESCRIPTION
**Pull request by zkwarlock**
Supports serialization and deserialization f32 and f64 types. This is done on client side using `Layout<number>`.

Example usage:

```ts
    await program.rpc.initialize(5000.11, {
      accounts: {
        myAccount: myAccount.publicKey,
        user: provider.wallet.publicKey,
        systemProgram: SystemProgram.programId,
      },
      signers: [myAccount],
    });
```
Pass float number directly, without BN. Negative numbers are supported.

### CODE REVIEW

**Comment by armaniferrante** [comment]
Awesome can we add an end to end test for this as well.

**Comment by Aadhinana** [comment]
> Awesome can we add an end to end test for this as well.

Do you mean like a separate dir in the `tests` folder or like a unit test or the travis CI thing not running?

**Comment by armaniferrante** [comment]
Please remove `tests/float/yarn.lock` and instead add the test to the newly added yarn workspace on master.

**Comment by Aadhinana** [comment]
> Please remove `tests/float/yarn.lock` and instead add `tests/float/` to the newly added yarn workspace on master.

Done sir! Please check!

File: tests/floats/.gitignore [hunk 1]
```diff
@@ -0,0 +1,6 @@
+
```
Comment (armaniferrante) [inline tests/floats/.gitignore:1]: Please remove this file.

Comment (Aadhinana) [reply tests/floats/.gitignore:1]: Sure!

Comment (Aadhinana) [reply tests/floats/.gitignore]: Does this require anything else to be done?

Comment (armaniferrante) [reply tests/floats/.gitignore]: Yes. CI fails.

**Comment by sraymor17** [comment]
Was this ever posted? I dont see /tests/float

**Comment by paul-schaaf** [comment]
we now have float support so Im closing this

**Pull request closed without merging**